### ISSUE TITLE
Two bugs regarding member with DW-NOMINATE score of 0

### ISSUE THREAD

**Issue by surf-solar** [voteview/WebVoteView#422]
Hi,

I've noticed that most House roll calls between the 84th and 95th Congresses have a glitch when you sort them by vote likelihood. During these Congresses, there was a very conservative Democratic (Dixiecrat?) representative from Georgia named John James Flynt Jr., and his DW-NOMINATE score is exactly 0 - (0, 1) with the two dimensions, since he's at the top of the second dimension. Not sure if that score is what causes this, but his presence in a roll call causes members to be sorted incorrectly, with Flynt's entry in between two groups of members. (see first image)
![image](https://github.com/voteview/WebVoteView/assets/104105173/d4af47e5-b7c8-4ef7-828e-924549fe8d55)

Also relating to Flynt, I've noticed you can only select him in the breakdown diagram if you click directly on his location in the diagram (which is at the extreme top). Selecting a portion of the diagram with your mouse will never pick him up. (see second image)
![image](https://github.com/voteview/WebVoteView/assets/104105173/39b4a72d-c408-4c4d-b277-3e4ec9af321e)

Both images are from [this roll call](https://voteview.com/rollcall/RH0900004).

**Comment by surf-solar** [comment]
Bumping since it's been a while. This glitch also shows up when someone doesn't have a score attached - such as in the 103rd, 110th, and 111th Congress on votes where delegates from the territories were counted, because they don't have a score attached (and their pages, except for Eleanor Holmes Norton, lead to errors). [This one, for example.](https://voteview.com/rollcall/RH1111493)

Also on members who don't have enough votes to be scored, such as [here](https://voteview.com/rollcall/RH1160952). Kwanza Hall (D-GA) and Paul Mitchell (I-MI) don't have a score so they mess up the order.

**Comment by JeffreyBLewis** [comment]
I don't think there is an issue to fix here.   Members are constrained to fall in the unit circle, so if a member is at "1"  or "-1" on one of the dimensions, they are at "0" on the other.   I will change the display of the scores so that instead of "0" or "1" which looks strange because they are rounded to a single digit, they display as "0.000" or "1.000".   

For members without scores, we omit them from the display when ordered by ideology and note how many are omitted for this reason.   Maybe it would be better to list them by name.  Let me know if you think that would be helpful, for now I am going to close this issue.

**Comment by surf-solar** [comment]
Appreciate the response. I've seen they're omitted when you sort by ideology, but not when you sort by the vote probability. And when you sort in a roll call where a member doesn't have a score (because of insufficient votes; most recently, see the votes with Adelita Grijalva) or (b) the member's dimension 1 score is exactly 0, the order of the votes messes up when it reaches the member in question. It also doesn't calculate the vote probability for the member, which makes sense in the first case but not in the second.

So with the votes that John Flynt (who I pointed out in the OP) partook in - and he was in Congress for over 20 years, so that's quite a few votes - none of his votes show a probability, and the order of vote probability resets when the table reaches him. I found another member who is at exactly 0 on dimension 1, but unlike with Flynt, he isn't at either extreme of dimension 2: [George Hancock (Federalist-VA, 1793-1797)](https://voteview.com/person/4026/george-hancock). You can select him since he isn't constrained by the edges, but because his dimension 1 score is exactly 0, no vote probability is calculated for any of his votes, and sorting by vote probability on any rollcall he voted in resets when it reaches him.

**Comment by JeffreyBLewis** [comment]
Yes.  What you say is true about folks without scores disappearing in the
ideological sort is true.  So, it might be better to list the "missing"
members in ideological sort at the bottom (where we now just point out that
there are two such members).   I had not noticed the issues with members
whose scores are 0.000.   I will look more into that.  Thanks for following
up.   I will just add that there shouldn't be anything magical about scores
of "0."  The 1st dim score has be 0 when the 2nd is 1 or -1, but a 1st dim
score of 0 can happen other times as well.  All of the scores are rounded
to three decimals and then displayed without trailing 0s.  Thus, "0", "1",
and "-1" look special.  However, there shouldn't be anything in the logic
that treats them differently.

**Comment by JeffreyBLewis** [comment]
There is logic in the generation of the predicted probability display that suppresses the predicted probabilities when the first dim score is exactly zero.  Because the legacy scores are rounded to three decimals in the DB, this happens a few times.  Here are those times:

bioname | chamber | nominate_dim1 | nominate_dim2 | min_cong | max_cong
-- | -- | -- | -- | -- | --
FLYNT, John James, Jr. | House | 0 | 1.000 | 84 | 95
HANCOCK, George | House | 0 | 0.215 | 3 | 4
HOWARD, Edgar | House | 0 | 1.000 | 68 | 73
MEADE, Wendell Howes | House | 0 | -0.440 | 80 | 80
MITCHELL, John Ridley | House | 0 | 1.000 | 72 | 75

I need to rule out any possible side effects of fixing this issue, but I think we should be able to remedy it easily.

**Comment by JeffreyBLewis** [comment]
I have patched up the predicted probabilities of folks whose 1st dim NOMINATE score is 0.  This should propagate to the production server's data within a few days.

**Comment by surf-solar** [comment]
Just checked and I can confirm all 5 members you listed have their probabilities showing up properly now. Thank you!

The probability sorting still messes up on members without scores though.
